pkg/telemetry/sqlite_metrics_db.go

```diff
@@ -120,25 +118,47 @@ func (b *sqliteMetricsDB) SaveOperationMetric(entry *OperationMetricsPayload) er
 		cliID:              entry.CliID,
 		commandStartTSMsec: strconv.FormatInt(entry.StartTime.UnixMilli(), 10),
 		commandEndTSMsec:   strconv.FormatInt(entry.EndTime.UnixMilli(), 10),
-		cspOrgID:           "", // TODO:(prkalle) to configure cspOrgID through telemetry plugin
-		accountNumber:      "", // TODO:(prkalle) to configure accountNumber through telemetry plugin
 		target:             entry.Target,
 		nameArg:            entry.NameArg,
 		endpoint:           entry.Endpoint,
 		flags:              entry.Flags,
 		exitStatus:         entry.ExitStatus,
-		isInternal:         true, // TODO:(prkalle) to configure isInternal through build options
+		isInternal:         entry.IsInternal,
 		error:              entry.Error,
 	}
 
-	_, err = db.Exec("INSERT INTO tanzu_cli_operations VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?);", row.cliVersion, row.osName, row.osArch, row.pluginName, row.pluginVersion, row.command, row.cliID, row.commandStartTSMsec, row.commandEndTSMsec, row.cspOrgID, row.accountNumber, row.target, row.nameArg, row.endpoint, row.flags, row.exitStatus, row.isInternal, row.error)
+	_, err = db.Exec("INSERT INTO tanzu_cli_operations VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?);", row.cliVersion, row.osName, row.osArch, row.pluginName, row.pluginVersion, row.command, row.cliID, row.commandStartTSMsec, row.commandEndTSMsec, row.target, row.nameArg, row.endpoint, row.flags, row.exitStatus, row.isInternal, row.error)
 	if err != nil {
 		return errors.Wrapf(err, "unable to insert clioperations row %v", row)
 	}
 
 	return nil
 }
 
+func (b *sqliteMetricsDB) GetRowCount() (int, error) {
+	err := AcquireTanzuMetricDBLock()
```

Comment:
Do we really need to gain lock to get count?!

Reply:
This is a safety net to avoid issues when users running the CLI concurrently. However I would file an issue to confirm if the sqlite native concurrency control would be good enough for this use case

Reply:
https://github.com/vmware-tanzu/tanzu-cli/issues/406